Wikipedia/Code/MWKImage.m

```diff
@@ -272,90 +218,34 @@ - (BOOL)isEqual:(id)object {
 }
 
 - (BOOL)isEqualToImage:(MWKImage*)image {
-    return self == image || [self.fileName isEqualToString:image.fileName];
+    return self == image || [[self.sourceURLString wmf_schemelessURL] isEqualToString:[image.sourceURLString wmf_schemelessURL]];
 }
 
 - (NSUInteger)hash {
     return [self.fileName hash];
 }
 
+- (BOOL)isPathEqualToImage:(MWKImage*)image {
+    return [[[self.sourceURLString wmf_schemelessURL] stringByDeletingLastPathComponent] isEqualToString:[[image.sourceURLString wmf_schemelessURL] stringByDeletingLastPathComponent]];
+}
+
 - (BOOL)isVariantOfImage:(MWKImage*)otherImage {
     // !!!: this might not be reliable due to underscore, percent encodings, and other unknowns w/ image filenames
     return [self.fileNameNoSizePrefix isEqualToString:otherImage.fileNameNoSizePrefix]
-           && ![self isEqualToImage:otherImage];
+           && [self isPathEqualToImage:otherImage];
```

Comment:
does path need to be checked if all images are on commons? I might need to alter the code I just added which only checks filenames.